packages/ui/lib/components/Collapse/Collapse.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react'
+import { Button } from '../Button/Button'
+import { FiChevronUp as ArrowUpIcon } from 'react-icons/fi'
+
+interface CollapseProps {
+  collapsed?: boolean
+  content: React.ReactNode
+  children: React.ReactNode
+  disabled?: boolean
+}
+export const Collapse = ({
+  children,
+  content,
+  disabled = false,
+  collapsed = false,
+}: CollapseProps) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const onCollapse = () => {
+    if (disabled) return
+    setIsOpen(!isOpen)
+  }
+
+  useEffect(() => {
+    setIsOpen(collapsed)
```

Comment:
@searchableguy the idea behind is to enable the ability to collapse/expand for a different source.

So that is why the component has `collapsed` props that change the internal state.
Any suggestion to do it better? member card for example can be expanded/closed from the filter bar

https://www.figma.com/file/pNN7ER1F4mdToFfk1bB0gu/App-v2?node-id=1587%3A28822

Reply:
Do you need `useEffect` here to change the state to default?

You can pass default value in the useState directly. Correct me if I'm missing something.

Since useState is a single line use. A user can use it outside the pure view component that this is and pass function and state down.